### DISCUSSION TITLE
Is there a way to have many fields that are not required, but required them when you submit the form?

### DISCUSSION THREAD

**Discussion by Drakota** [react-hook-form/react-hook-form#2120] [Q&A]
Hi!
Right now I have a form that can submit partial data, so I've put all my fields without field validation `{ required: true }` and that works, but I want to change the validation schema when I send the form with another button. One button can send partial data, but another one needs all fields required.

How can I achieve that in the same form?

**Comment by bluebill1049** [comment]
sounds like you want to do validation at the submit level and `setError`
